commit 1081312f95be1a0fc093556d789e482d3ab0f5b0
Author: Michal Hocko <mhocko@suse.cz>
Date:   Fri Feb 22 16:35:41 2013 -0800

    memcg: cleanup mem_cgroup_init comment
    
    We should encourage all memcg controller initialization independent on a
    specific mem_cgroup to be done here rather than exploit css_alloc
    callback and assume that nothing happens before root cgroup is created.
    
    Signed-off-by: Michal Hocko <mhocko@suse.cz>
    Acked-by: Johannes Weiner <hannes@cmpxchg.org>
    Cc: Tejun Heo <htejun@gmail.com>
    Cc: KAMEZAWA Hiroyuki <kamezawa.hiroyu@jp.fujitsu.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/mm/memcontrol.c b/mm/memcontrol.c
index 604eddc58665..d4f0a831c9e6 100644
--- a/mm/memcontrol.c
+++ b/mm/memcontrol.c
@@ -6830,10 +6830,12 @@ static void __init enable_swap_cgroup(void)
 #endif
 
 /*
- * The rest of init is performed during ->css_alloc() for root css which
- * happens before initcalls.  hotcpu_notifier() can't be done together as
- * it would introduce circular locking by adding cgroup_lock -> cpu hotplug
- * dependency.  Do it from a subsys_initcall().
+ * subsys_initcall() for memory controller.
+ *
+ * Some parts like hotcpu_notifier() have to be initialized from this context
+ * because of lock dependencies (cgroup_lock -> cpu hotplug) but basically
+ * everything that doesn't depend on a specific mem_cgroup structure should
+ * be initialized from here.
  */
 static int __init mem_cgroup_init(void)
 {